### ISSUE TITLE
Fixed yielding rate in the generator

### ISSUE THREAD

**Issue by Adk2001tech** [aniketmaurya/chitra#129]
### Problem
There is a **limited data yield** in the [generator](https://github.com/aniketmaurya/chitra/blob/1543805f1401c571e516e47ab1c8a83b93dd657c/chitra/datagenerator.py#L195) which stops data iteration (hence no data will now yield by generator) in 1 complete cycle. Limiting developer to reassess the same data twice in a loop.

### Solution and Alternatives 
Making the generator yields **inexhaustible data** and simultaneously adding **training steps** in model training.

**Comment by aniketmaurya** [comment]
Thanks for raising the issue @Adk2001tech 😀
Would you like to create a PR for this?

**Comment by Adk2001tech** [comment]
@aniketmaurya please assign this to me✋. I am working on it😃.
Thanks

**Comment by Adk2001tech** [comment]
I have some questions @aniketmaurya 

- does [validation_data](https://github.com/aniketmaurya/chitra/blob/156e97557a2b9e84fe13ff1747c115aaaa68c181/chitra/trainer.py#L223) is an Object of class [Dataset](https://github.com/aniketmaurya/chitra/blob/156e97557a2b9e84fe13ff1747c115aaaa68c181/chitra/datagenerator.py#L99) ? (can necessarily be the object or can be just a simple tuple of (image, label) pairs)
- same for [train_data](https://github.com/aniketmaurya/chitra/blob/156e97557a2b9e84fe13ff1747c115aaaa68c181/chitra/trainer.py#L394) in class [Learner](https://github.com/aniketmaurya/chitra/blob/156e97557a2b9e84fe13ff1747c115aaaa68c181/chitra/trainer.py#L378) .

As it is note mentioned in code comments.
Thanks

**Comment by aniketmaurya** [comment]
Hi @Adk2001tech, both `validation_data` and `train_data` are of type [tf.data](https://www.tensorflow.org/guide/data)

**Comment by aniketmaurya** [comment]
btw I am also thinking if we migrate from TF to PyTorch. We will keep the `Datagenerator` the same. If you want to help in this may be we can discuss this over Discord- https://discord.com/invite/TdnAfDw3kB

**Comment by Adk2001tech** [comment]
> Hi @Adk2001tech, both `validation_data` and `train_data` are of type [tf.data.Dataset](https://www.tensorflow.org/guide/data)

sure 👍
